charmrepo/charmstore.go

```diff
@@ -71,7 +74,58 @@ func (s *CharmStore) Get(curl *charm.URL) (charm.Charm, error) {
 
 // Latest implements Interface.Latest.
 func (s *CharmStore) Latest(curls ...*charm.URL) ([]CharmRevision, error) {
-	return nil, notImplemented
+	if len(curls) == 0 {
+		return nil, nil
+	}
+
+	// Prepare the request to the charm store.
+	urls := make([]string, len(curls))
+	ids := make([]string, len(curls))
+	for i, curl := range curls {
+		url := curl.WithRevision(-1).String()
+		urls[i] = url
+		ids[i] = "id=" + url
+	}
+	path := "/meta/any?include=id-revision&include=hash256&" + strings.Join(ids, "&")
+
+	// Execute the request and retrieve results.
+	var results map[string]struct {
```

Comment:
Nice suggestion! Done.
csclient.BulkMeta is also a good idea, I'll investigate that later.